File: Euclid/Network/Streams/Util/KeyValueEntry.cs
namespace Euclid.Network.Streams.Util
{
    public class KeyValueEntry
    {
        #region Private variables

        private string m_Key;
        private string m_Value;
        private string m_Delimiter;

        #endregion

        #region Fields

        public string Key
        {
            get { return m_Key; }
        }

        public string Value
        {
            get { return m_Value; }
        }

        public string Delimiter
        {
            get { return m_Delimiter; }
        }

        #endregion

        #region Constructor

        public KeyValueEntry(string key, object value, object delimiter)
        {
            m_Key = key;
            m_Value = value.ToString();
            m_Delimiter = delimiter.ToString();
        }

        #endregion
    }
}

File: euclid/src/main/java/euclid/network/streams/util/KeyValueEntry.java
package euclid.network.streams.util;

public final class KeyValueEntry {

    private final String key;
    private final String value;
    private final String delimiter;

    public KeyValueEntry(String key, Object value, Object delimiter) {
        this.key = key;
        this.value = value.toString();
        this.delimiter = delimiter.toString();
    }

    public String getKey() {
        return key;
    }

    public String getValue() {
        return value;
    }

    public String getDelimiter() {
        return delimiter;
    }
}
